src/go/capture/buffer/buffer.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2021 The Magma Authors.
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree.
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package buffer
+
+import (
+	"sync"
+
+	"github.com/magma/magma/src/go/protos/magma/capture"
+)
+
+//go:generate go run github.com/golang/mock/mockgen -source=buffer.go -package mock_buffer -destination mock_buffer/mock_buffer.go Buffer
+
+// buffer interface defines functions of Buffer and is used for generating mocks.
+type buffer interface {
```

Comment:
this feels reversed, `Buffer` should be the interface and `buffer` the implementation

for a simple struct like this, it's OK to not have an interface (testers can instantiate the concrete impl if necessary; it's entirely in memory and doesn't have any dependencies)

Reply:
i can drop the interface but Buffer needs to be exported for use in the capture server